### ISSUE TITLE
Bug: Panic when running `moment`'s test suite

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno#24408]
```sh
Platform: macos aarch64
Version: 1.44.4
Args: ["/Users/marvinh/dev/denoland/deno/target/debug/deno", "run", "--unstable-ffi", "--unstable-fs", "--unstable-webgpu", "--node-modules-dir", "-A", "/Users/marvinh/dev/oss/moment/node_modules/node-qunit/lib/child.js", "grunt", "/Users/marvinh/dev/oss/moment/node_modules/grunt/bin/grunt", "test", ...]

thread 'main' panicked at /Users/marvinh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/deno_core-0.292.0/ops.rs:54:5:
op op_node_ipc_write was not marked as #[op2(reentrant)], but re-entrantly invoked op op_apply_source_map
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Steps to reproduce

1. Clone https://github.com/moment/moment
2. Run `npm i`
3. Run `DENO_FUTURE=1 deno task test`

Version: Deno 1.44.4 (git 496ea5903 )

**Comment by J053Fabi0** [comment]
I don't know if it's relevant, but moment is not working since Deno 1.44.3.

1.44.2:
![image](https://github.com/denoland/deno/assets/12930717/00b0513d-05f7-49c9-8767-d260b0717c04)

1.44.3 and 1.44.4:
![image](https://github.com/denoland/deno/assets/12930717/d9c16329-6630-4c19-9a43-15d9ab94873c)

**Comment by marvinhagemeister** [comment]
@J053Fabi0 It sounds like you're not getting the same panic error. Can you file a separate issue for that? That way we can much more effectively triage issues and work on them in parallel.

**Comment by birkskyum** [comment]
With 1.45.5 I get:

```sh
Running "qtest" task
Deprecation warning: testing deprecation
Arguments: 
Error
    at file:///Users/admin/repos/deno-kitchensink/moment/build/umd/test/moment/deprecate.js:175:25
    at Object.<anonymous> (file:///Users/admin/repos/deno-kitchensink/moment/build/umd/test/moment/deprecate.js:193:9)
    at runTest (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/qunit/qunit/qunit.js:2697:35)
    at Test.run (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/qunit/qunit/qunit.js:2680:9)
    at file:///Users/admin/repos/deno-kitchensink/moment/node_modules/qunit/qunit/qunit.js:2951:16
    at processTaskQueue (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/qunit/qunit/qunit.js:2271:26)
    at file:///Users/admin/repos/deno-kitchensink/moment/node_modules/qunit/qunit/qunit.js:2275:13
    at eventLoopTick (ext:core/01_core.js:207:9)
error: Uncaught TypeError: table.push is not a function
    at Object.print.globalSummary (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/node-qunit/lib/log.js:203:11)
    at file:///Users/admin/repos/deno-kitchensink/moment/node_modules/node-qunit/lib/testrunner.js:189:40
    at Function.each (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/underscore/underscore-node-f.cjs:1328:7)
    at file:///Users/admin/repos/deno-kitchensink/moment/node_modules/node-qunit/lib/testrunner.js:187:19
    at complete (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/node-qunit/lib/testrunner.js:74:9)
    at ChildProcess.<anonymous> (file:///Users/admin/repos/deno-kitchensink/moment/node_modules/node-qunit/lib/testrunner.js:104:17)
    at ChildProcess.emit (ext:deno_node/_events.mjs:387:28)
    at handleMessage (ext:deno_node/internal/child_process.ts:1090:14)
    at processTicksAndRejections (ext:deno_node/_next_tick.ts:30:15)
    at runNextTicks (ext:deno_node/_next_tick.ts:71:3)
```

**Comment by littledivy** [comment]
Seems to work with `--unsafe-unstable-proto`. See #24986 

```
$ cat deno.json
{
  "unstable": ["unsafe-proto"]
}

$ DENO_FUTURE=1 deno task test

global summary:
┌───────┬───────┬────────────┬────────┬────────┬─────────┐
│ Files │ Tests │ Assertions │ Failed │ Passed │ Runtime │
├───────┼───────┼────────────┼────────┼────────┼─────────┤
│ 1     │ 3871  │ 162849     │ 0      │ 162849 │ 4220    │
└───────┴───────┴────────────┴────────┴────────┴─────────┘

Running "exec:typescript-test" (exec) task

> moment@2.30.1 typescript-test
> cross-env node_modules/typescript/bin/tsc --project typing-tests


Running "exec:ts3.1-typescript-test" (exec) task

> moment@2.30.1 ts3.1-typescript-test
> cross-env node_modules/typescript3/bin/tsc --project ts3.1-typing-tests


Done.
```

**Comment by bartlomieju** [comment]
The panic is no longer reproducible on Deno 2.7.4.

```
$ deno task test
Task test grunt test
Loading "Gruntfile.js" tasks...ERROR
>> ReferenceError: module is not defined
```

The original crash (`op op_node_ipc_write was not marked as #[op2(reentrant)], but re-entrantly invoked op op_apply_source_map`) no longer occurs — Deno no longer panics.

There is a remaining non-panic issue: `Gruntfile.js` uses `module.exports` (CJS) but Deno loads it as ESM, causing `module is not defined`. This is a separate CJS detection issue, not the panic reported here.

Closing the panic as fixed. Please open a new issue if the CJS detection problem is a blocker.
